src/js/media-controller.js

```diff
@@ -159,14 +159,63 @@ class MediaController extends MediaContainer {
       MEDIA_ENTER_PIP_REQUEST: () => {
         const media = this.media;
 
-        if (!document.pictureInPictureEnabled) return;
+        if (!document.pictureInPictureEnabled) {
+          console.warn('MediaChrome: Picture-in-picture is not enabled');
+          // Placeholder for emitting a user-facing warning
+          return;
+        };
+
+        if (!media.requestPictureInPicture) {
+          console.warn('MediaChrome: The current media does not support picture-in-picture');
+          // Placeholder for emitting a user-facing warning
+          return;
+        }
 
         // Exit fullscreen if needed
         if (document[fullscreenApi.element]) {
           document[fullscreenApi.exit]();
         }
 
-        media.requestPictureInPicture();
+        const warnNotReady = (err) => {
+          console.warn('MediaChrome: The media is not ready for picture-in-picture. It must have a readyState > 0.');
+        };
+
+        media.requestPictureInPicture().catch(err => {
+          // InvalidStateError, readyState == 0 (Not ready)
+          if (err.code === 11) {
+            // We can assume the viewer wants the video to load, so attempt to
+            // if we can rely on readyState and preload
+            // Only works in Chrome currently. Safari doesn't allow triggering
+            // in an event listener. Also requires readyState == 4.
+            // Firefox doesn't do PiP yet.
+            if (media.readyState === 0 && media.preload === 'none') {
```

Comment:
question(non-blocking): Is there a reason we're only retrying if `media.preload === 'none'`? Will this edge case not arise if someone requests PIP when `media.readyState === 0` and `media.preload` is `"metadata"` or `"auto"` (aka attempted to enter PIP faster than the src's metadata load)?

Reply:
> attempted to enter PIP faster than the src's metadata load

I see how the logic is open for that but honestly that feels like such a rare situation that I don't think it's worth complicating this any further to support it. If it actually comes up I'll update it then.